data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -231,7 +232,13 @@ public void nonNullWrite(int rowId, String data, ColumnVector output) {
 
     @Override
     public void nonNullWrite(int rowId, ByteBuffer data, ColumnVector output) {
-      ((BytesColumnVector) output).setRef(rowId, data.array(), 0, data.array().length);
+      if (data.hasArray()) {
+        ((BytesColumnVector) output).setRef(rowId, data.array(),
+                data.arrayOffset() + data.position(), data.remaining());
```

Comment:
There was a line length breaking linter error here. Not sure what the expected way to break up the argument list is, so I chose something that seemed to make sense and that passed the linter.